### ISSUE TITLE
Cannot map container port to 8080 (not available)

### ISSUE THREAD

**Issue by corrieb** [vmware/vic#2006]
root@ubuntu-1604-vmware:~/vic/src/github.com/vmware/vic-n# docker run -d -p 8080:80 nginx
4c85c24058372f10334a419c077c333bbd70546f2abbef069ac01f93ea4b5c71
docker: Error response from daemon: error mapping ports: port 8080 is not available.

Speculation on vic-dev is that the port layer is published on port 8080. Many customer demos will expect to be able to use port 8080 to publish on, so this is a major UX issue.

Also related to #1986 which is a different problem but has the same solution

Our internal services should use obscure port numbers that we document as being unavailable. Would be nice if we can detect these port numbers and say "this is in use internally by the VCH" rather than just "not available".

**Comment by mdubya66** [comment]
I consider this a show stopper. Marking as p0.

**Comment by hmahmood** [comment]
The port layer uses 8080, as well as the attach server. We need to move both to obscure ports.

**Comment by hmahmood** [comment]
```hmahmood@ubuntu:~/go/src/github.com/vmware/vic$ grep 8080 lib/* cmd/* pkg/*  -r
lib/install/management/appliance.go:				"-port-layer-port=8080",
lib/install/management/appliance.go:				"--port=8080",
lib/install/validate/config.go:		Port:      8080, // serialOverLANPort
lib/install/validate/config.go:		msg := "Firewall must permit 8080/tcp outbound to use VIC"
lib/install/validate/config.go:		log.Warning("Firewall must permit 8080/tcp outbound if firewall is reenabled")
lib/portlayer/portlayer.go:	SerialOverLANPort = 8080
lib/portlayer/exec/handle.go:	serialOverLANPort  = 8080
cmd/imagec/imagec.go:	DefaultPortLayerHost = "localhost:8080"
cmd/tether/attach_test.go:	testServer := attach.NewAttachServer("", 8080)```

**Comment by hmahmood** [comment]
2377 is still available: http://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.txt

I am going to use that since we are already using 2378 for vicadmin.

**Comment by hickeng** [comment]
I'd like to revert this change. What we've just done is require that almost all VIC deployments have ESX firewall disabled, modified on each host boot, or manually create and deploy a VIB to all hosts - this just isn't a viable stance.

We can either:
a. find a different port that is also enabled outbound by default on ESX 6.5
b. continue to use 8080 and doc the conflict
c. install a VIB during VCH create that configures the firewall and hope that no new hosts are added to the cluster after the fact.

**Comment by hmahmood** [comment]
I don't agree that using 8080 is a good idea -- to many users default to that after port 80. And I also do not want to use another port that is designated for something else. Installing a VIB of course is only viable until another host is added, as @hickeng points out.

Instead of reverting this change, I believe we should seriously consider adding 2377 to the default firewall port rulesets that ESX ships with. That port is actually available universally, and we should reserve it for VIC.

**Comment by hickeng** [comment]
@hmahmood that works for the release after 6.5, but isn't a viable solution for the currently shipping versions that we need to support.

The other possibility is that we figure out how to make the multiple interfaces work so we don't have this conflict at all - that should function and completely eliminates this problem.
I didn't understand the issue you referenced in #2056 about the client IP address and issues with it changing when DHCP - the iptables rules should be using `-i external` to specify incoming traffic rather than an interface IP address.

**Comment by hmahmood** [comment]
We have already documented this issue: https://github.com/vmware/vic/blob/master/doc/user_doc/vic_installation/ts_firewall_error.md . This is a little bit of a hurdle to adoption, but I really don't think we have any other good option other than to document this. 

Having two ip addresses on the same subnet on two NIC's is something that is brittle, and difficult to pull off. Not recommended as per my discussion with @mlh78750 . Also see here: http://serverfault.com/questions/336021/two-network-interfaces-and-two-ip-addresses-on-the-same-subnet-in-linux . I did attempt it on my own also, and could not get it to work.

As for iptables -- the problem is putting the ip address of the the client/external interface in the rules when using DHCP for those addresses. If the address changes, deleting that rule is not possible since we don't have the old address, since the iptables command will no longer match the rule. As you pointed out, matching by interface is sufficient, and works in case the ip address changes.

**Comment by mdubya66** [comment]
we need to have a discussion with @karthik-narayan as there seems to be disagreement on how to proceed with it.

**Comment by hmahmood** [comment]
Looks like a VIB to open up the firewall to port 2377 is the most viable solution. This requires some additional work around signing the VIB so that the ESX will install it, in addition to possibly installing it during vic-machine runs. Per discussion with @mdubya66, we are iceboxing this for the upcoming GA release. For now, users can open the firewall manually; instructions are [here](https://github.com/vmware/vic/blob/master/doc/user_doc/vic_installation/ts_firewall_error.md)

**Comment by hickeng** [comment]
Remaining firewall config work will be addressed under #3643
